File: 2265-partition-array-according-to-given-pivot/partition-array-according-to-given-pivot.java
class Solution {
    public int[] pivotArray(int[] nums, int pivot) {
        int n = nums.length ;
        int l = 0 , r = n - 1 ;
        int left = 0 , right = n - 1;
        int[] res = new int[n];
        while (l < n) 
        {
            if (nums[l] < pivot)
            {
                res[left] = nums[l] ;
                left++ ;
            }
            if (nums[r] > pivot)
            {
                res[right] = nums[r] ;
                right-- ;
            }
            l++ ;
            r --;
        }
        while( left <= right )
        {
            res[left] = pivot ;
            res[right] = pivot ;
            left ++ ;
            right -- ;
        }
        return res;
        
    }
}

File: 2265-partition-array-according-to-given-pivot/partition-array-according-to-given-pivot.py
class Solution:
    def pivotArray(self, nums: List[int], pivot: int) -> List[int]:
        l , left = 0 , 0
        r , right = len(nums) - 1 , len(nums) - 1
        res = [0] * len(nums)

        while l < len(nums):
            if nums[l] < pivot:
                res[left] = nums[l]
                left += 1
            if nums[r] > pivot:
                res[right] = nums[r]
                right -= 1
            l += 1
            r -= 1
        while left <= right:
            res[left] = pivot
            left += 1

        return res 


        